geode-dunit/src/main/java/org/apache/geode/cache30/MultiVMRegionTestCase.java

```diff
@@ -8284,45 +8284,46 @@ public void run() {
+          Map r2Contents = vm2.invoke(MultiVMRegionTestCase::getCCRegionContents);
+          Map r3Contents = vm3.invoke(MultiVMRegionTestCase::getCCRegionContents);
 
-      try {
-        for (int i = 0; i < 10; i++) {
-          String key = "cckey" + i;
-          assertThat(r2Contents.get(key)).describedAs(
-              "1 region contents are not consistent for i " + key).isEqualTo(r1Contents.get(key));
-          assertThat(r3Contents.get(key)).describedAs(
-              "2 region contents are not consistent for i " + key).isEqualTo(r2Contents.get(key));
-          for (int subi = 1; subi < 3; subi++) {
-            String subkey = key + "-" + subi;
-            if (r1Contents.containsKey(subkey)) {
-              assertThat(r2Contents.get(subkey)).describedAs(
-                  "3 region contents are not consistent for " + subkey).isEqualTo(
-                      r1Contents.get(subkey));
-              assertThat(r3Contents.get(subkey)).describedAs(
-                  "4 region contents are not consistent for " + subkey).isEqualTo(
-                      r2Contents.get(subkey));
-            } else {
-              boolean condition1 = !r2Contents.containsKey(subkey);
-              assertThat(condition1).describedAs("5 r2contents").isTrue();
-              boolean condition = !r3Contents.containsKey(subkey);
-              assertThat(condition).describedAs("6 r3contents").isTrue();
+          try {
+            for (int i = 0; i < 10; i++) {
+              String key = "cckey" + i;
+              assertThat(r2Contents.get(key)).describedAs(
+                  "r2 contents are not consistent with r1 for " + key)
+                  .isEqualTo(r1Contents.get(key));
+              assertThat(r3Contents.get(key)).describedAs(
+                  "r3 contents are not consistent with r2 for " + key)
+                  .isEqualTo(r2Contents.get(key));
+              for (int subi = 1; subi < 3; subi++) {
+                String subkey = key + "-" + subi;
+                if (r1Contents.containsKey(subkey)) {
+                  assertThat(r2Contents.get(subkey)).describedAs(
+                      "r2 contents are not consistent with r1 for subkey " + subkey).isEqualTo(
+                          r1Contents.get(subkey));
+                  assertThat(r3Contents.get(subkey)).describedAs(
+                      "r3 contents are not consistent with r2 for subkey " + subkey).isEqualTo(
+                          r2Contents.get(subkey));
+                } else {
+                  assertThat(r2Contents.containsKey(subkey))
+                      .describedAs("r2 contains subkey " + subkey + " that r1 does not").isFalse();
+                  assertThat(r3Contents.containsKey(subkey))
+                      .describedAs("r3 contains subkey " + subkey + " that r1 does not").isFalse();
+                }
+              }
             }
+          } catch (Exception e) {
```

Comment:
You shouldn't need to catch and rethrow, as `ThrowingRunnable` throws `Throwable`. I think Awaitility will print the stack trace, too.